### PR TITLE
feat: Refactor Config out of Runtime

### PR DESCRIPTION
**Pull request by evgenykuzyakov**
Move `RuntimeConfig` from `Runtime` initialization to `ApplyState` object.
This makes `Runtime` completely stateless and allows to execute transitions based on different configs.
It allows to upgrade `RuntimeConfig` with the new fees based on the protocol version for the current block.

This change should be NOOP and doesn't change the protocol. 

NEP: https://github.com/nearprotocol/NEPs/issues/120
Issue: #3158

# Test plan:

- [X] CI
- [x] http://nayduck.eastus.cloudapp.azure.com:3000/#/run/553

### CODE REVIEW

File: neard/src/runtime.rs [hunk 1]
```diff
@@ -113,6 +114,7 @@ impl EpochInfoProvider for SafeEpochManager {
 /// TODO: this possibly should be merged with the runtime cargo or at least reconciled on the interfaces.
 pub struct NightshadeRuntime {
     genesis_config: GenesisConfig,
+    genesis_runtime_config: Arc<RuntimeConfig>,
```
Comment (bowenwang1996) [inline neard/src/runtime.rs:117]: why do we need to do this? runtime config is part of `genesis_config`

Comment (evgenykuzyakov) [reply neard/src/runtime.rs:117]: You'd have to clone it every time to wrap into `Arc`

**Review by bowenwang1996 — APPROVED** [review]

**Review by willemneal — APPROVED** [review]

File: neard/src/runtime.rs (hunk 1 above)
Comment (frol) [reply neard/src/runtime.rs:117]: It seems that you could implement it with lifetimes, but Arc is fine

**Review by frol — APPROVED** [review]

File: core/runtime-configs/src/lib.rs [hunk 2]
```diff
@@ -45,6 +47,13 @@ impl RuntimeConfig {
             account_creation_config: AccountCreationConfig::default(),
         }
     }
+
+    pub fn from_protocol_version(
```
Comment (nearmax) [inline core/runtime-configs/src/lib.rs:51]: This needs a TODO with a GH issue linked to it.

File: neard/src/runtime.rs (hunk 1 above)
Comment (nearmax) [reply neard/src/runtime.rs:118]: Lifetimes tend to poison code. It is better if we first solidify our codebase with smart pointers and then switch to references.

**Review by nearmax — APPROVED** [review]
Some nits.
It is unclear what `RuntimeConfig::from_protocol_version` is for.

**Pull request merged**